tests/spin_targets.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python
+#
+#    Copyright 2024 ARM Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""
+Module for testing different targets.
+Forked from tests/test_target.py.
+"""
+
+import os
+from unittest import TestCase
+from devlib import AndroidTarget, LinuxTarget, LocalLinuxTarget, QEMUTargetRunner
+from devlib.utils.misc import get_random_string
+from devlib.exception import TargetStableError
+
+
+class TestReadTreeValues(TestCase):
+    """ Class testing Target.read_tree_values_flat() """
+
+    def test_read_multiline_values(self):
+        """ Test Target.read_tree_values_flat() """
+
+        def run_test(target):
+            data = {
+                'test1': '1',
+                'test2': '2\n\n',
+                'test3': '3\n\n4\n\n',
+            }
+
+            print(f'{method_name}: target={target.__class__.__name__} os={target.os} ' \
+                  f'hostname={target.hostname}')
+
+            dirname = os.path.join(target.working_directory,
+                                   f'devlib-test-{get_random_string(12)}')
+            print(f'{method_name}: creating {dirname}...')
+            target.makedirs(dirname)
+
+            try:
+                for key, value in data.items():
+                    path = os.path.join(dirname, key)
+                    print(f'{method_name}: writing {value!r} to {path}...')
+                    target.write_value(path, value, verify=False,
+                                       as_root=target.conn.connected_as_root)
+
+                print(f'{method_name}: reading values from target...')
+                raw_result = target.read_tree_values_flat(dirname)
+                result = {os.path.basename(k): v for k, v in raw_result.items()}
+            except Exception as ex:
+                raise TargetStableError('Error while processing the tree!') from ex
+            finally:
+                print(f'{method_name}: removing {dirname}...')
+                target.remove(dirname)
+
+            self.assertEqual({k: v.strip()
+                              for k, v in data.items()},
+                             result)
+
+            print(f'{method_name}: removing {target.working_directory}...')
+            target.remove(target.working_directory)
+
+        def create_targets():
+            targets = []
```

Comment:
looks like this whole function could just be:
```
return [
    AndroidTarget(connection_settings={'device': 'emulator-5554'},
                                     working_directory='/data/local/tmp/devlib-target'),
...
]
```

instead of loads of append() calls